src/idc/idc.c

```diff
@@ -312,6 +324,9 @@ void idc_cmd(struct idc_msg *msg)
 	case iTS(IDC_MSG_PREPARE_D0ix):
 		idc_prepare_d0ix();
 		break;
+	case iTS(IDC_MSG_AMS):
+		idc_process_async_msg(msg->header & 0xFFFF);
```

Comment:
this means you can have up to 64K slots?..

Reply:
I don't see why not, other than the 16 bit slot part of IDC format.

Reply:
My bad, that limit would be 16, as the slot parameter here acts as a bitmask. Anyway, the transport layer should not check logical validity of the data. It's the processing's role to supply and handle correct values. And if we only need 4 of them, that's ok. IDC should only verify if the header is compliant with the format (so check header type and mask the value, anything else?)

Reply:
```
idc_process_async_msg(msg->header & 0xFFFF);
process_incoming_message(msg->header & 0xFFFF);
ams_task_add_slot_to_process(ams_task, msg->header & 0xFFFF);
	ams_task->pending_slots |= BIT(msg->header & 0xFFFF);
```
that doesn't look like it's treated as a bitmask to me